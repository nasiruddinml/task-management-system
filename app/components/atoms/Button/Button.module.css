/*
  buttons
*/

.button {
  background: var(--color-primary);
  border: 0;
  border-radius: 0.25rem;
  color: var(--color-white);
  cursor: pointer;
  font: inherit;
  font-weight: 500;
  height: 3rem;
  line-height: 3rem;
  padding: 0 2rem;

  /* grid */
  align-items: center;
  display: grid;
  gap: 0.75rem;
  grid-auto-flow: column;
}
.button:hover {
  background: var(--color-primary-500);
}
.button i {
  font-size: 1.25rem;
}

.button.icon {
  border-radius: 50%;
  justify-content: center;
  overflow: hidden;
  padding: 0;
  width: 3rem;
}

.button.link {
  background: none;
  color: var(--color-primary);
  text-decoration: underline;
  justify-content: start;
  padding: 0;
  height: 1.5rem;
  line-height: 1.25rem;
}
.button.link:hover {
  color: var(--color-primary-500);
}
.button.link i {
  font-size: inherit;
}
