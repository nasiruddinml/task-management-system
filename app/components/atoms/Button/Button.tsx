import React, { ReactNode, MouseEvent } from 'react';
import styles from './Button.module.css';

export const Button = ({
  children,
  ...rest
}: React.DetailedHTMLProps<
  React.ButtonHTMLAttributes<HTMLButtonElement>,
  HTMLButtonElement
>) => {
  return (
    <button
      {...rest}
      className={`${styles.button} ${
        rest.className === 'link' ? styles.link : rest.className
      }`}
    >
      {children}
    </button>
  );
};

export default Button;
