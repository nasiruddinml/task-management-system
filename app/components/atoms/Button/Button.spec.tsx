import { render, screen } from '@testing-library/react';
import Button from '@components/atoms/Button/Button';

describe('Button Component test', () => {
  it('renders a text', () => {
    render(<Button>I am a button</Button>);

    const buttonElement = screen.getByText('I am a button');

    expect(buttonElement).toBeInTheDocument();
  });

  it('renders button unchanged', () => {
    const { container } = render(<Button>I am a button</Button>);
    expect(container).toMatchSnapshot();
  });
});
