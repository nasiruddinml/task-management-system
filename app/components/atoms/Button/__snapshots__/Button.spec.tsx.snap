// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button Component test renders button unchanged 1`] = `
<div>
  <button
    class="button undefined"
  >
    I am a button
  </button>
</div>
`;
